test/runtests.jl

```diff
@@ -1155,6 +1193,53 @@ Base.similar(bc::Broadcast.Broadcasted{Broadcast.ArrayStyle{MyArray}}, ::Type{El
 
     @test identity.(StructArray(x=StructArray(a=1:3)))::StructArray == [(x=(a=1,),), (x=(a=2,),), (x=(a=3,),)]
     @test (x -> x.x.a).(StructArray(x=StructArray(a=1:3))) == [1, 2, 3]
+    @test identity.(StructArray(x=StructArray(x=StructArray(a=1:3))))::StructArray == [(x=(x=(a=1,),),), (x=(x=(a=2,),),), (x=(x=(a=3,),),)]
+    @test (x -> x.x.x.a).(StructArray(x=StructArray(x=StructArray(a=1:3)))) == [1, 2, 3]
+
+    @testset "ambiguity check" begin
+        function _test(a, b, c)
+            if a isa StructArray || b isa StructArray || c isa StructArray
+                d = @inferred a .+ b .- c
+                @test d == collect(a) .+ collect(b) .- collect(c)
+                @test d isa StructArray
+            end
+        end
+        testset = Any[StructArray([1;2+im]),
+                    1:2, 
+                    (1,2),
+                    StructArray(@SArray [1 1+2im]),
+                    (@SArray [1 2])
+                    ]
+        for aa in testset, bb in testset, cc in testset
```

Comment:
Not sure if this is the culprit, but the broadcast tests add substantially to the testing time of StructArrays.

They certainly are very thorough (which is great!), but I was wondering whether there is a simple way to make them faster while keeping them as thorough as they are now. If not that's no big deal, it's definitely better to be thorough than to be fast. It could also be that testing with StaticArrays is inherently a bit slow due to compilation issues.

Reply:
With 129b904 the time for broadcast test is reduced to 50% (39s -> 19s on my PC.)